src/huggingface_hub/hf_file_system.py

```diff
@@ -0,0 +1,431 @@
+                if "/" in revision_in_path:
+                    revision_in_path, path_in_repo = revision_in_path.split("/", 1)
+                else:
+                    path_in_repo = ""
+                revision_in_path = unquote(revision_in_path)
+                revision = _align_revision_in_path_with_revision(revision_in_path, revision)
+                repo_and_revision_exist, err = self._repo_and_revision_exist(repo_type, repo_id, revision)
+                if not repo_and_revision_exist:
+                    raise FileNotFoundError(path) from err
+            else:
+                repo_id_with_namespace = "/".join(path.split("/")[:2])
+                path_in_repo_with_namespace = "/".join(path.split("/")[2:])
+                repo_id_without_namespace = path.split("/")[0]
+                path_in_repo_without_namespace = "/".join(path.split("/")[1:])
+                repo_id = repo_id_with_namespace
+                path_in_repo = path_in_repo_with_namespace
+                repo_and_revision_exist, err = self._repo_and_revision_exist(repo_type, repo_id, revision)
+                if not repo_and_revision_exist:
+                    if isinstance(err, (RepositoryNotFoundError, HFValidationError)):
+                        repo_id = repo_id_without_namespace
+                        path_in_repo = path_in_repo_without_namespace
+                        repo_and_revision_exist, _ = self._repo_and_revision_exist(repo_type, repo_id, revision)
+                        if not repo_and_revision_exist:
+                            raise FileNotFoundError(path) from err
+                    else:
+                        raise FileNotFoundError(path) from err
+        else:
+            repo_id = path
+            path_in_repo = ""
+            if "@" in path:
+                repo_id, revision_in_path = path.split("@", 1)
+                revision_in_path = unquote(revision_in_path)
+                revision = _align_revision_in_path_with_revision(revision_in_path, revision)
+            repo_and_revision_exist, _ = self._repo_and_revision_exist(repo_type, repo_id, revision)
+            if not repo_and_revision_exist:
+                raise NotImplementedError("Acces to repositories lists is not implemented.")
+
+        revision = revision if revision is not None else DEFAULT_REVISION
+        return ResolvedPath(repo_type, repo_id, revision, path_in_repo)
+
+    def invalidate_cache(self, path=None):
+        if not path:
+            self.dircache.clear()
+            self._repository_type_and_id_exists_cache.clear()
+        else:
+            path = self._strip_protocol(path)
+            path = self.resolve_path(path).unresolve()
+            while path:
+                self.dircache.pop(path, None)
+                path = self._parent(path)
+
+    def _open(
+        self,
+        path: str,
+        mode: str = "rb",
+        revision: Optional[str] = None,
+        **kwargs,
+    ):
+        if mode == "ab":
```

Comment:
Appending without binary mode is not supported either, right?

```suggestion
        if mode == "ab" or mode == "a":
```